src/config/env.py

```diff
@@ -0,0 +1,105 @@
+"""
+env.py
+
+This module is responsible for loading, parsing, and validating the application's
+environment configuration from the `env.yaml` file. It uses Pydantic to ensure that
+the configuration adheres to expected formats and types.
+
+Functions:
+- create_default_config: Creates a default configuration file if it doesn't exist.
+
+Classes:
+- EnvConfig: Loads the configuration and exposes it as Python objects.
+- RemoteNode: Represents a remote node configuration with validation.
+- Config: Represents the overall configuration structure with validation.
+
+Usage:
+- Initialize an instance of EnvConfig to load the configuration.
+- Access configuration values via properties (e.g., env.remote_nodes).
+"""
+
+from pathlib import Path
+from typing import Dict, List
+
+import yaml
+from pydantic import BaseModel, HttpUrl, ValidationError
+
+ENV_PATH = Path(__file__).resolve().parent.parent.parent / "env.yaml"
+
+
+def create_default_config():
+    """Creates a default configuration file if it doesn't exist."""
+    default_config = """
+# This file contains environment-specific configurations for the application.
+# It is git-ignored to keep local secrets safe.
+
+# The configuration stores Ethereum RPC node details.
+remote_nodes:
```

Comment:
Ah, of course, if you're adding comments I can see why templating would be helpful. 

We could use a combination of python objects holding the default values and a jinja2 template. We can add this later though!